build.js

```diff
@@ -0,0 +1,105 @@
+import fs from 'fs';
+
+import copy from 'esbuild-plugin-copy';
+import esbuild from "esbuild";
+import { sassPlugin } from 'esbuild-sass-plugin';
+
+import { cockpitCompressPlugin } from './pkg/lib/esbuild-compress-plugin.js';
+import { cockpitPoEsbuildPlugin } from './pkg/lib/cockpit-po-plugin.js';
+import { cockpitRsyncEsbuildPlugin } from './pkg/lib/cockpit-rsync-plugin.js';
+import { cleanPlugin } from './pkg/lib/esbuild-cleanup-plugin.js';
+import { eslintPlugin } from './pkg/lib/esbuild-eslint-plugin.js';
+import { stylelintPlugin } from './pkg/lib/esbuild-stylelint-plugin.js';
+import { replace } from 'esbuild-plugin-replace';
+import { esbuildStylesPlugins } from './pkg/lib/esbuild-common.js';
+
+const production = process.env.NODE_ENV === 'production';
+const watchMode = process.env.ESBUILD_WATCH === "true" || false;
+// linters dominate the build time, so disable them for production builds by default, but enable in watch mode
+const lint = process.env.LINT ? (process.env.LINT !== '0') : (watchMode || !production);
+/* List of directories to use when resolving import statements */
+const nodePaths = ['pkg/lib'];
+const outdir = 'dist';
+
+// Obtain package name from package.json
+const packageJson = JSON.parse(fs.readFileSync('package.json'));
+
+const getTime = () => new Date().toTimeString()
+        .split(' ')[0];
+
+const cwd = process.cwd();
+
+const context = await esbuild.context({
+    ...!production ? { sourcemap: "external" } : {},
+    bundle: true,
+    entryPoints: ["./src/index.js"],
+    external: ['*.woff', '*.woff2', '*.jpg', '*.svg', '../../assets*'], // Allow external font files which live in ../../static/fonts
+    legalComments: 'external', // Move all legal comments to a .LEGAL.txt file
+    loader: {
+        ".js": "jsx",
+        ".py": "text",
+    },
+    minify: production,
+    nodePaths,
+    outdir,
+    target: ['es2020'],
+    plugins: [
+        cleanPlugin(),
+        ...lint ? [
+            stylelintPlugin({ filter: new RegExp(cwd + '\/src\/.*\.(css?|scss?)$') }),
```

Comment:
Let's make that the default in cockpit's pkg/lib/ as well, to be consistent with eslintPlugin, and avoid repetition in all projects. (not a blocker for this PR, of course)